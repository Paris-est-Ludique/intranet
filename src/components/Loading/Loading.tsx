import styles from './styles.module.scss'

export default function Loading(): JSX.Element {
  return (
    <div className={styles.loading}>
      <p>Loading...</p>
    </div>
  )
}
