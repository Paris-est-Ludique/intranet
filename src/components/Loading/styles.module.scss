.loading {
  padding: 0 15px;
}
