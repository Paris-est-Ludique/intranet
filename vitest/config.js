export default {
  preset: 'ts-jest',
  rootDir: '../',
  testEnvironment: 'node',
  setupFilesAfterEnv: ['<rootDir>/jest/setup.ts'],
  collectCoverageFrom: [
    'src/app/**/*.tsx',
    'src/pages/**/*.tsx',
    '!src/pages/**/index.tsx',
    'src/components/**/*.tsx',
    'src/store/**/*.ts',
    '!src/store/index.ts',
    '!src/store/rootReducer.ts',
  ],
  moduleNameMapper: {
    '.*\\.(css|scss|sass)$': '<rootDir>/jest/styleMock.ts',
    '.*\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
            '<rootDir>/jest/assetMock.ts',
  },
  globals: {
    __DEV__: true,
    __CLIENT__: true,
    __SERVER__: false,
    __LOCAL__: false,
    __REGISTER_DISCORD_COMMANDS__: false,
    __TEST__: true,
  },
  maxConcurrency: 50,
  maxWorkers: 1,
}
