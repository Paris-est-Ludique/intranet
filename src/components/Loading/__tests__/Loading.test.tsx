import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import Loading from '../Loading'

describe('<Loading />', () => {
  it('renders', () => {
    const tree = render(
      <MemoryRouter>
        <Loading />
      </MemoryRouter>,
    ).container.firstChild

    expect(tree).toMatchSnapshot()
  })
})
