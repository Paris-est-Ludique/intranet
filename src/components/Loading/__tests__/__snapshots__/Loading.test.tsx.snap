// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Loading /> > renders 1`] = `
<div>
  <p>
    Loading...
  </p>
</div>
`;
